src/common.js

```diff
@@ -157,8 +157,8 @@ module.exports = function setup(env) {
   function enable(namespaces) {
     createDebug.save(namespaces);
 
-    createDebug.names = [];
-    createDebug.skips = [];
+    createDebug.names = createDebug.names || [];
```

Comment:
Why are these two necessary? Won't `createDebug.names` always be present?

Reply:
namespace support need to keep old names, if `createDebug.names = []`, would break here: https://github.com/visionmedia/debug/pull/506/files/439e358fd48adb95d373e525d1dc1cffac4d20b0#diff-ec54818fc97b81da761d4b169c52c0ecR80